pkg/ebpf/bytecode/runtime/telemetry.go

```diff
@@ -0,0 +1,132 @@
+//go:build linux_bpf
+// +build linux_bpf
+
+package runtime
+
+import (
+	"time"
+	"fmt"
+	"strings"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/DataDog/nikos/types"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+// CompilationResult enumerates runtime compilation success & failure modes
+type CompilationResult int
+
+const (
+	notAttempted CompilationResult = iota
+	compilationSuccess
+	kernelVersionErr  // nolint:deadcode,unused
+	verificationError
+	outputDirErr
+	outputFileErr
+	newCompilerErr // nolint:deadcode,unused
+	compilationErr
+	resultReadErr
+	headerFetchErr
+	compiledOutputFound
+	inputHashError
+)
+
+// CompilationTelemetry is telemetry collected per-program when attempting runtime compilation
+type CompilationTelemetry struct {
+	compilationEnabled  bool
+	compilationResult   CompilationResult
+	compilationDuration time.Duration
+	headerFetchResult   kernel.HeaderFetchResult
+}
+
+func newCompilationTelemetry() CompilationTelemetry {
+	return CompilationTelemetry{
+		compilationEnabled: false,
+		compilationResult:  notAttempted,
+		headerFetchResult:  kernel.NotAttempted,
+	}
+}
+
+// CompilationEnabled returns whether or not runtime compilation has been enabled
+func (tm *CompilationTelemetry) CompilationEnabled() bool {
+	return tm.compilationEnabled
+}
+
+// CompilationResult returns the result of runtime compilation
+func (tm *CompilationTelemetry) CompilationResult() int32 {
+	return int32(tm.compilationResult)
+}
+
+// CompilationDurationNS returns the duration of runtime compilation
+func (tm *CompilationTelemetry) CompilationDurationNS() int64 {
+	return tm.compilationDuration.Nanoseconds()
+}
+
+// KernelHeaderFetchResult returns the result of kernel header fetching
+func (tm *CompilationTelemetry) KernelHeaderFetchResult() int32 {
+	return int32(tm.headerFetchResult)
+}
+
+// SubmitTelemetry sends telemetry using the provided statsd client
+func (tm *CompilationTelemetry) SubmitTelemetry(filename string, statsdClient statsd.ClientInterface) {
```

Comment:
@paulcacheux This function replaces the [SendMetrics function](https://github.com/DataDog/datadog-agent/blob/89f00e4df66327fe2f054f78e833ba6480619167/pkg/ebpf/bytecode/runtime/runtime_compilation_helpers.go#L82-L106) which was previously responsible for submitting telemetry related to runtime compiled constants. If that's going to be an issue (ie because the metric name will be different now), let me know, I can add the old function back in as well.

Reply:
This should not be an issue no

Reply:
The only thing that I would like would be to either add a check for `statsdClient == nil` here or at call sites to handle the case where there is not client to submit to (for example in tests)

Reply:
Good call. Done!